openpathsampling/pathmover.py

```diff
@@ -546,225 +538,105 @@ def __call__(self, trial):
 
         return trials
 
-    def _shoot(self, shooting_point, ensemble):
-        """Implementation of the shooting
-
-        Parameters
-        ----------
-        shooting_point : ShootingPoint
-            the initial shooting point instance containing a reference to the
-            initial trajectory
-        ensemble : Ensemble
-            the ensemble for which the trajectory is to be created. This defines
-            the stopping criterion
-
-        Returns
-        -------
-        ShootingPoint
-            the final shooting point referencing the final trajectory
-        """
-        return shooting_point
-
-
-class ForwardShootMover(ShootMover):
-    """A forward shooting sample generator
-    """
-    def _shoot(self, shooting_point, ensemble):
-        shoot_str = "Shooting {sh_dir} from frame {fnum} in [0:{maxt}]"
-        logger.info(shoot_str.format(fnum=shooting_point.index,
-                                     maxt=len(shooting_point.trajectory)-1,
-                                     sh_dir="forward",
-                                    ))
-
-        # Run until one of the stoppers is triggered
-        partial_trajectory = self.engine.generate(
-            shooting_point.snapshot.copy(),
-            running = [
-                paths.PrefixTrajectoryEnsemble(
-                    ensemble,
-                    shooting_point.trajectory[0:shooting_point.index]
-                ).can_append
-            ]
-        )
-
-        trial_trajectory = \
-            shooting_point.trajectory[0:shooting_point.index] + \
-            partial_trajectory
-
-        trial_point = paths.ShootingPoint(
-            shooting_point.selector,
-            trial_trajectory,
-            shooting_point.index
-        )
+    def _make_forward_trajectory(self, trajectory, shooting_index):
+        initial_snapshot = trajectory[shooting_index].copy()
```

Comment:
Question: Why do we make a copy of the snapshot? It does not change the configuration, velocities or reversed state, right? Then it is virtually the same as the old one and we don't need a copy. I am not sure, but I think that the engine cannot alter the given snapshot so it will create a redundant snapshot everytime. Has this something todo with potential two-way shooting?